mesonbuild/backend/backends.py

```diff
@@ -638,9 +638,9 @@ def as_meson_exe_cmdline(self, exe: T.Union[str, mesonlib.File, build.BuildTarge
                 return es.cmd_args, ''
             args: T.List[str] = []
             if capture:
-                args += ['--capture', str(capture)]
+                args += ['--capture', capture]
             if feed:
-                args += ['--feed', str(feed)]
+                args += ['--feed', feed]
```

Comment:
When you think about it, this is terrifying. The type annotations say it's a bool, so we cast it to string in the event that it's a literal True? Weird.

This got introduced in f6f35aa906c8ebac21af07c17449df65d08aeaeb with a lot of other code, but probably it was a confusion based on CustomTarget taking capture as a boolean and creating an ExecutableSerialisation as `capture=ofilenames[0] if target.capture else None`. Nice...

Reply:
I've been working on the backend off and on, there's still almost 400 errors in ninjabackend.py. Three of them are:

> mesonbuild/backend/ninjabackend.py:1170:57: error: Argument "capture" to "as_meson_exe_cmdline" of "Backend" has incompatible type "str"; expected "bool | None"  [arg-type]
> mesonbuild/backend/ninjabackend.py:1171:54: error: Argument "feed" to "as_meson_exe_cmdline" of "Backend" has incompatible type "str"; expected "bool | None"  [arg-type]
> -mesonbuild/backend/ninjabackend.py:2654:65: error: Argument "capture" to "as_meson_exe_cmdline" of "Backend" has incompatible type "str"; expected "bool | None"  [arg-type]

This change fixes those.

Reply:
Merged via 28e6d2be9621dcd1056d6e386576510fd9cd8860, will take a look at the rest later...